Base/QTCore/qSlicerExtensionsManagerModel.h

```diff
@@ -399,6 +399,13 @@ public slots:
   /// Returns false if waitForCompletion is set to true and metadata cannot be retrieved.
   bool updateExtensionsMetadataFromServer(bool force=false, bool waitForCompletion=false);
 
+
+  /// \brief Enable/disable the use of offset for querying metadata from server.
+  ///
+  /// By default, default offset value is used to query the metadata by batch.
```

Comment:
Describe here the expected server behavior for the flag enabled and disabled. For enabled: query is repeated with increasing offset value, until 0 items are returned. For disabled I can just guess: one query is expected to return all items, offset value is ignored.

Reply:
When useOffset is false, the value -1 is passed meaning that all extensions are returns.

By default, the first query returns the default offset size (50 from memory) ... then the next offset is based on the number of extension already retrieved. 

I am wondering if we should instead always pass -1 and simplify the logic ? 

Though is nice to have a more robust approach that doesn't require an amount of memory depending on the number of extensions ...

Reply:
Based on my limited experience with web APIs, querying with offsets is a common pattern, probably because it is more robust (requires less memory, processing time, etc.) and more scalable (works for any number of items), allows progress reporting, etc. The client-side logic for multiple queries is quite simple and already implemented, so I think it is fine to keep its. It is nice to have the single-query option, too. So, we just need to explain in the documentation why there are these two different query modes, how they work, what are the advantages/disadvantages.